scripts/start_update.sh

```diff
@@ -5,8 +5,199 @@
+# The directory in the chroot environment containing the source update files.
+UPDATE_ROOTFS_SOURCE="/mnt/update_rootfs_source"
+
+update_rootfs_archive=""
+
+
+usage()
+{
+    echo "Usage: ${0} [OPTIONS] <TOOLBOX_MOUNT> <UPDATE_MOUNT> <TARGET_STORAGE_DEVICE>"
+    echo "This is the update entry point script, it is responsible for setting up the"
+    echo "environment in which the update toolbox can be used to configure and update"
+    echo "the firmware."
+    echo "  -h Print this help text and exit"
+}
+
+cleanup()
+{
+    if [ -d "${TOOLBOX_MOUNT}/${UPDATE_ROOTFS_SOURCE}" ]; then
+        echo "Cleaning up, removing '${TOOLBOX_MOUNT}/${UPDATE_ROOTFS_SOURCE}/' files."
+        rm -rf "${TOOLBOX_MOUNT:?}/${UPDATE_ROOTFS_SOURCE:?}"/*
+    fi
+
+    if [ -f "${UPDATE_MOUNT}/${update_rootfs_archive}" ]; then
+        echo "Cleaning up, removing '${UPDATE_MOUNT}/${update_rootfs_archive}'."
+        rm "${UPDATE_MOUNT:?}/${update_rootfs_archive:?}"
+    fi
+
+    if grep -q "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}" /proc/mounts; then
+        echo "Cleaning up, unmount: '${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}'."
+        umount "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}"
+    fi
+}
+
+extract_update_rootfs()
+{
+    update_rootfs_pattern="rootfs*.tar.xz"
+
+    echo "Looking for update rootfs files with the pattern '${update_rootfs_pattern}' on '${UPDATE_MOUNT}/${update_rootfs_archive}'."
+    nr_files="$(find "${UPDATE_MOUNT}" -maxdepth 1 -iname "${update_rootfs_pattern}" | wc -l)"
+    if [ "${nr_files}" -eq 0 ]; then
+        echo "Error, update failed: no update rootfs with the pattern '${update_rootfs_pattern}' found on '${UPDATE_MOUNT}'."
+        return 1
+    fi
+
+    if [ "${nr_files}" -gt 1 ]; then
+        echo "Error, update failed: multiple update filesystem archives found on '${UPDATE_MOUNT}'."
+        return 1
+    fi
+
+    # shellcheck disable=SC2086
+    # Disabled because file globbing is needed here
+    update_rootfs_archive="$(basename "${UPDATE_MOUNT}/"${update_rootfs_pattern})"
+    echo "Found '${update_rootfs_archive}', attempting to extract to '${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}'."
+    if ! tar -tJf "${UPDATE_MOUNT}/${update_rootfs_archive}" > /dev/null 2> /dev/null; then
+        echo "Error, update failed: ${UPDATE_MOUNT}/${update_rootfs_archive} is corrupt"
+        return 1
+    fi
+
+    if [ ! -d "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}" ]; then
+        echo "Error, update failed: source update directory: '${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}' does not exist."
+        return 1
+    fi
+
+    temp_folder="$(mktemp -d)"
+    if ! mount --bind "${temp_folder}" "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}"; then
+        echo "Error, update failed: temp source update directory: '${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}' cannot be mounted."
+        return 1
+    fi
+
+    if ! tar xvfJ "${UPDATE_MOUNT}/${update_rootfs_archive}" -C "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}" \
```

Comment:
The verbose flag on tar causes each filename to be printed to the serial console, which will slow down the extraction on the target. As the extraction will be faster then the serial port and buffers will run full.

Reply:
Wait, it's verbose, but the output is directed to /dev/null?